### PR TITLE
BugFix - Sync Nested Folder

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

**How to Test?**

The sync folder action should sync all files within the selected folder, not just those one level below.

### CODE REVIEW

File: app/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java [hunk 1]
```diff
@@ -414,23 +413,21 @@ private void classifyFileForLaterSyncOrDownload(OCFile remoteFile, OCFile localF
 
 
     private void prepareOpsFromLocalKnowledge() throws OperationCancelledException {
-        List<OCFile> children = getStorageManager().getFolderContent(mLocalFolder, false);
+        List<OCFile> children = getStorageManager().getAllFilesRecursivelyInsideFolder(mLocalFolder);
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java:416]: If we want to have nested folder sync, we should start with SynchronizeFolderOperation on folders, but not only on files.

**Review by tobiasKaminsky — CHANGES_REQUESTED** [review]
SynchronizeFolderOperation needs to be used.

**Comment by alperozturk96** [comment]
Duplication of https://github.com/nextcloud/android/pull/14166

**Comment by pretubedongit** [comment]
Why is this closed? It's not a duplication of #14166. This is about nested folder or level 1 files only. It's still present in the latest version.

**Pull request closed without merging**